cdc-accelerator/feature/permissionGroups.js

```diff
@@ -45,16 +48,83 @@ export default class PermissionGroups extends PartnerFeature {
         this.copyFileFromSrcToBuild(srcFeaturePath, PermissionGroups.PERMISSIONGROUP_FILE_NAME)
     }
 
-    async deploy(directory) {
-        console.log('deploy called')
+    async deploy(partnerDirectory, siteInfo) {
+        const buildFeatureDirectory = path.join(partnerDirectory, this.getName())
+        const buildFileName = path.join(buildFeatureDirectory, `${this.getName()}.json`)
+        const fileContent = fs.readFileSync(buildFileName, { encoding: 'utf8' })
+
+        if (!fileContent || !fileContent.length) {
+            throw new Error(`Invalid file: ${buildFileName}`)
+        }
+        const parsedContent = JSON.parse(fileContent)
+
+        let keys = Object.keys(parsedContent)
+        let response
+        for (let ids of keys) {
+            if (parsedContent[ids].scope) {
+                response = await this.deployPermissionGroup(siteInfo, ids, parsedContent[ids], this.credentials)
+            }
+            response = await this.updatePermissionGroup(siteInfo, ids, parsedContent[ids], this.credentials)
+            if (response.errorCode !== 0) {
+                throw new Error(JSON.stringify(response))
+            }
+        }
+    }
+
+    async deployPermissionGroup(siteInfo, groupId, config, credentials) {
+        return await this.setPermissionRequest(siteInfo.dataCenter, siteInfo.partnerId, groupId, config, credentials.userKey, credentials.secret)
+    }
+    async updatePermissionGroup(siteInfo, groupID, config, credentials) {
+        return await this.updatePermissionGroupRequest(siteInfo.dataCenter, siteInfo.partnerId, groupID, config, credentials)
     }
     async getPermissionGroups(dataCenter, partnerID, credentials) {
         const url = `https://admin.${dataCenter}.gigya.com/admin.getGroups`
-        const response = await client.post(url, this.#setPermissionGroupsParameters(partnerID, credentials.userKey, credentials.secret)).catch((error) => error)
+        const response = await client.post(url, this.#getPermissionGroupsParameters(partnerID, credentials.userKey, credentials.secret))
+        return response.data
+    }
+    async setPermissionRequest(dataCenter, partnerID, groupId, config, userKey, secret) {
+        const url = `https://admin.${dataCenter}.gigya.com/admin.createGroup`
+        const response = await client.post(url, this.#setPermissionGroupsParameters(partnerID, userKey, secret, groupId, config))
         return response.data
     }
 
-    #setPermissionGroupsParameters(partnerID, userKey, secret) {
+    async updatePermissionGroupRequest(dataCenter, partnerID, groupID, config, credentials) {
+        const url = `https://admin.${dataCenter}.gigya.com/admin.updateGroup`
+        const response = await client.post(url, this.#updatePermissionGroupsParameters(partnerID, groupID, config, credentials.userKey, credentials.secret))
+        return response.data
+    }
+    #setPermissionGroupsParameters(partnerID, userKey, secret, groupID, config) {
```

Comment:
The method #setPermissionGroupsParameters is really needed, can't be used #updatePermissionGroupsParameters instead?